extensions/azure/cosmos/transfer-process-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/transfer/store/cosmos/CosmosTransferProcessStoreIntegrationTest.java

```diff
@@ -180,6 +164,28 @@ void nextForState() throws InterruptedException {
                 .noneMatch(p -> p.getId().equals(id3));
     }
 
+    @Test
+    void nextForState_leaseByAnotherHolderExpired() {
+        String id1 = UUID.randomUUID().toString();
+        var tp = createTransferProcess(id1, TransferProcessStates.INITIAL);
+        TransferProcessDocument item = new TransferProcessDocument(tp, partitionKey);
+        Duration leaseDuration = Duration.ofSeconds(5);
+        item.acquireLease("another-connector", leaseDuration);
+        container.upsertItem(item);
+
+        List<TransferProcess> processesBeforeLeaseBreak = store.nextForState(TransferProcessStates.INITIAL.code(), 10);
+        assertThat(processesBeforeLeaseBreak).isEmpty();
+
+        await()
+                .atMost(Duration.ofSeconds(20))
+                .pollInterval(Duration.ofSeconds(1))
+                .pollDelay(leaseDuration) //give the lease time to expire
+                .untilAsserted(() -> {
+                    List<TransferProcess> processesAfterLeaseBreak = store.nextForState(TransferProcessStates.INITIAL.code(), 10);
+                    assertThat(processesAfterLeaseBreak).hasSize(1);
```

Comment:
nit: same as contract negotiation assertions.
```suggestion
                    assertThat(processesAfterLeaseBreak).hasSize(1).allSatisfy(transferProcess -> assertThat(transferProcess).usingRecursiveComparison().isEqualTo(tp));
```